Loop/Managers/DeviceDataManager.swift

```diff
@@ -331,14 +329,32 @@ final class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverD
         // If we don't yet have pump status, or it's old, poll for it.
         if latestReservoirValue == nil || latestReservoirValue!.startDate.timeIntervalSinceNow <= -pumpStatusAgeTolerance {
             readPumpData { (result) in
+                let nsPumpStatus: NightscoutUploadKit.PumpStatus?
                 switch result {
                 case .Success(let (status, date)):
                     self.updateReservoirVolume(status.reservoir, atDate: date, withTimeLeft: nil)
+                    let battery = BatteryStatus(voltage: status.batteryVolts, status: BatteryIndicator(batteryStatus: status.batteryStatus))
+                    nsPumpStatus = NightscoutUploadKit.PumpStatus(clock: date, pumpID: status.pumpID, iob: nil, battery: battery, suspended: status.suspended, bolusing: status.bolusing, reservoir: status.reservoir)
                 case .Failure:
                     self.troubleshootPumpCommsWithDevice(device)
+                    nsPumpStatus = nil
+                }
+                
+                if nsPumpStatus != nil {
+                    // Always upload new pump data
+                    self.nightscoutDataManager.uploadDeviceStatus(nsPumpStatus)
+                    self.lastDeviceStatusUpload = NSDate()
+                } else {
+                    // Only upload if it's been a while
+                    if self.lastDeviceStatusUpload == nil || -(self.lastDeviceStatusUpload!.timeIntervalSinceNow) >= NSTimeInterval(minutes: 4) {
+                        self.nightscoutDataManager.uploadDeviceStatus(nsPumpStatus)
```

Comment:
Yes, it shows uploader battery and shows that the Loop app is running and able to upload to NS, even if loop() is progressing for some reason (more on this later).